common/djangoapps/third_party_auth/models.py

```diff
@@ -880,12 +895,37 @@ def get_config(self):
         conf['x509certMulti'] = {'signing': public_keys}
         conf['x509cert'] = ''
         conf['url'] = sso_url
+        # This block determines which SAML configuration should be used during authentication
+        if self.saml_configuration:
+            # If this provider has a specific configuration assigned to it
+            # Use the toggle to determine behavior
+            if USE_LATEST_SAML_CONFIG.is_enabled():
+                # Look for the latest enabled configuration with the same slug
```

Comment:
- Move some of this code before the toggle. Use a try-block to ensure that if any errors occur while doing new processing, it doesn't mess with the legacy behavior.
- Add observability for new vs old when the toggle is off, and possibly whether they would return the same value.
- Use the toggle to determine whether to return the new or old result.

Again, I didn't have time to actually review the code.

Reply:
@ktyagiapphelix2u: I thought you said this was refactored, but maybe the PR doesn't yet have the latest changes?